packages/components/src/components/CodeEditor.tsx

```diff
@@ -61,10 +62,15 @@ export const CodeEditor: FC<CodeEditorProps> = ({
   oneLine = false,
   showGutter = false,
   errorLocations = [],
+  project,
 }) => {
+  if (oneLine && height)
+    throw Error("Either `height` or `oneLine` may be set.");
+
   const editor = useRef<HTMLDivElement>(null);
   const editorView = useRef<EditorView | null>(null);
-  const languageSupport = squiggle();
+  const languageSupport = useMemo(() => squiggle(project)(), [project]);
```

Comment:
What's the point of having an extra function call here?

Reply:
I needed for `(project)` invokation to be lazy, and for some reason I didn't go and change language support constructor all together — iirc it was requiring a lambda in some other place. I'll untangle that